src/test/java/core/basesyntax/impl/ReaderServiceImplTest.java

```diff
@@ -0,0 +1,49 @@
+package core.basesyntax.impl;
+
+import static org.junit.Assert.assertEquals;
+
+import core.basesyntax.service.ReaderService;
+import core.basesyntax.service.ReaderServiceImpl;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.junit.Before;
+import org.junit.Test;
+
+public class ReaderServiceImplTest {
+    private static final String NAME_EMPTY_FILE =
+            "src/test/java/core/basesyntax/impl/emptyFile.csv";
+    private static final String FILE_NAME_TO_NULL = null;
+    private static final String NAME_FILE_WITH_INFO =
+            "src/test/java/core/basesyntax/impl/CorrectData.csv";
+    private static ReaderService readerService;
+    private static final List<String> VALID_LIST = List.of(
+            "type", "fruit", "quantity",
+            "b", "banana", "20",
+            "b", "apple", "100");
+
+    @Before
+    public void beforeClass() {
+        readerService = new ReaderServiceImpl();
+    }
+
+    @Test(expected = RuntimeException.class)
+    public void realizePattern_readEmptyFile_notOk() {
```

Comment:
```suggestion
    public void read_readEmptyFile_notOk() {
```
same in other method names